webrtc.html

```diff
@@ -3088,25 +3101,96 @@ <h2>Methods</h2>
+                        <p>If <var>type</var> is <code>"offer"</code>, and
+                        <var>sdp</var> is not the empty string and not equal to
                         <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
-                        slot.</p>
+                        slot, then return a promise <a>rejected</a> with a newly
+                        <a data-link-for="exception" data-lt="create">created</a>
+                        <code>InvalidModificationError</code> and abort these
+                        steps.</p>
+                      </li>
+                      <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                        <p>If <var>type</var> is <code>"answer"</code> or
+                        <code>"pranswer"</code>, and <var>sdp</var> is not the
+                        empty string and not equal to <var>connection</var>'s
+                        <a>[[\LastCreatedAnswer]]</a> slot, then return a
+                        promise <a>rejected</a> with a newly
+                        <a data-link-for="exception" data-lt="create">created</a>
+                        <code>InvalidModificationError</code> and abort these
+                        steps.</p>
+                      </li>
+                      <li>
+                        <p>If <var>sdp</var> is the empty string, and
+                          <var>type</var> is <code>"offer"</code>, then run the
+                          following sub steps:</p>
+                        <ol>
+                          <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                            <p>Set <var>sdp</var> to the value of
+                            <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
+                            slot.</p>
+                          </li>
+                          <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                            <p>If <var>sdp</var> is the empty string, or if it
+                            no longer accurately represents the system state of
+                            <var>connection</var>, or if <var>connection</var>'s
+                            configured indentity provider has changed since
+                            <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
+                            was created, then let <var>p</var> be the result of
+                            <a href="#dfn-create-an-offer">creating an offer</a>
+                            with <var>connection</var>, and return the result of
+                            <a href="https://w3ctag.github.io/promises-guide/#transforming-by">
+                            transforming</a> <var>p</var> with a fulfillment
+                            handler that returns the result of
+                            <a href="#set-description">setting the RTCSessionDescription</a>
+                            indicated by its first argument.</p>
```

Comment:
"indicated by its first argument"? I have trouble parsing this sentence.
Should it say "setting the RTCSessionDescription with the created offer"?

Reply:
This language predates this PR. I also find it a bit odd, but the word "RTCSessionDescription" here refers to the input, not the set methods of a similar name.

I'm working on another PR that looks to clean this up a bit FWIW.